### PR TITLE
task 1 commit

### PR DESCRIPTION
**Pull request by 16liaod**
Basic implementation of task 1 for meeting 1, not sure if this is all we need to do yet.

### CODE REVIEW

File: Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs [hunk 1]
```diff
@@ -20,20 +27,78 @@ public class TestClass : Character
...
+                retlist.Add(new int[tilex, tiley]);
+            }
+        }
+        return retlist;
+    }
+    public override void ondeathhandler() //need to wait on this one to decide how summoning is implemented etc
+    {
+        if(currentHealth <= 0)
+        {
+            GameObject.Find("GameManager").GetComponent<GameManager>().player1Units.Remove(this.gameObject);
```
Comment (awc71403) [inline Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs]: Change to gameManager.GetSingleton()

File: Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs [hunk 2]
```diff
@@ -20,20 +27,78 @@ public class TestClass : Character
...
+        {
+            return false;
+        }
     }
 
-    public override List<int[,]> GetAttackRange() {
+    public override List<int[,]> GetAttackRange()
+    {
+        List<int[,]> retlist = new List<int[,]>();
+        foreach(GameObject tile in GameObject.Find("GameManager").GetComponent<GameManager>().mapArray)
```
Comment (awc71403) [inline Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs]: Change to gameManager.GetSingleton()

File: Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs [hunk 3]
```diff
@@ -20,20 +27,78 @@ public class TestClass : Character
...
+        if(currentHealth <= 0)
+        {
+            GameObject.Find("GameManager").GetComponent<GameManager>().player1Units.Remove(this.gameObject);
+        }
+    }
+
+    public override void attack(GameObject target)
+    {
+        //TODO: implement this
+        target.GetComponent<TestClass>().TakeDamage(damage);
```
Comment (awc71403) [inline Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs]: Shouldn't this be target.GetComponent<Character>().TakeDamage(damage)?

File: Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs [hunk 4]
```diff
@@ -20,20 +27,78 @@ public class TestClass : Character
...
+            int tilex = tile.GetComponent<TileBehavior>().xPosition;
+            int tiley = tile.GetComponent<TileBehavior>().yPosition;
+            if (IsInRange(tilex, tiley))
+            {
+                retlist.Add(new int[tilex, tiley]);
+            }
+        }
+        return retlist;
+    }
+    public override void ondeathhandler() //need to wait on this one to decide how summoning is implemented etc
```
Comment (awc71403) [inline Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs:127]: Couldn't this be called under TakeDamage? All units are summoned from PlaceCharacterOnTile in GameManager and are put in a GameObject list. One for player1 and another or player2.

**Review by awc71403 — CHANGES_REQUESTED** [review]

File: Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs (hunk 3 above)
Comment (16liaod) [reply Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs]: takedamage is a function inside the test class file i believe

File: Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs (hunk 2 above)
Comment (16liaod) [reply Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs]: included in next merge request

File: Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs (hunk 1 above)
Comment (16liaod) [reply Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs]: included in next merge request

File: Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs (hunk 4 above)
Comment (16liaod) [reply Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs:127]: will include in future push

**Pull request merged**